featuretools/primitives/standard/aggregation_primitives.py

```diff
@@ -882,3 +882,69 @@ def frequency_counter(column_vals):
             assert len(freq) == len(self.baseline_vals), err_msg
             return freq
         return frequency_counter
+
+
+class FirstNotNull(AggregationPrimitive):
+    """Determines the first value in a list that is not null.
+
+    Examples:
+        >>> first = FirstNotNull()
+        >>> first([None, np.NaN, 1, 2, 3, 4, 5, None])
+        1.0
+    """
+    name = "first_not_null"
+    input_types = [Variable]
+    return_type = None
+    stack_on_self = False
+    description_template = "the first non-null instance of {}"
+
+    def get_function(self, agg_type=Library.PANDAS):
+        def pd_first_not_null(x):
+            idx = x.first_valid_index()  # will return None if all Nones
+            return x.loc[idx] if idx is not None else None
+
+        return pd_first_not_null
+
+
+class LastNotNull(AggregationPrimitive):
+    """Determines the first value in a list that is not null.
+
+    Examples:
+        >>> last = LastNotNull()
+        >>> last([None, np.NaN, 1, 2, 3, 4, 5, None])
+        5.0
+    """
+    name = "last_not_null"
+    input_types = [Variable]
+    return_type = None
+    stack_on_self = False
+    description_template = "the last non-null instance of {}"
+
+    def get_function(self, agg_type=Library.PANDAS):
+        def pd_first_not_null(x):
+            idx = x.last_valid_index()  # will return None if all Nones
+            return x.loc[idx] if idx is not None else None
+
+        return pd_first_not_null
+
+
+class ModeNotNull(AggregationPrimitive):
+    """Determines the first value in a list that is not null.
+
+    Examples:
+        >>> last = LastNotNull()
+        >>> last([None, np.NaN, 1, 2, 3, 4, 5, None])
+        5.0
+    """
+    name = "last_not_null"
+    input_types = [Variable]
+    return_type = None
+    stack_on_self = False
+    description_template = "the last non-null instance of {}"
+
+    def get_function(self, agg_type=Library.PANDAS):
+        def pd_first_not_null(x):
+            idx = x.last_valid_index()  # will return None if all Nones
+            return x.loc[idx] if idx is not None else None
+
```

Comment:
That doesn't fit with Mode

Reply:
Oops, this wasn't supposed to get here, I copy-pasted FirstNotNull, and then figured out that the normal Mode filters out Nones anyway.